packages/cosmwasm/src/cw1subkeycosmwasmclient.ts

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Coin } from "@cosmjs/launchpad";
+
+import { Cw1CosmWasmClient } from "./cw1cosmwasmclient";
+import { ExecuteResult } from "./signingcosmwasmclient";
+import { Expiration } from "./types";
+
+export interface AllowanceResult {
+  readonly balance: readonly Coin[];
+  readonly expires: Expiration;
+}
+
+export interface Cw1SubkeyPermissions {
+  readonly delegate: boolean;
+  readonly redelegate: boolean;
+  readonly undelegate: boolean;
+  readonly withdraw: boolean;
+}
+
+export class Cw1SubkeyCosmWasmClient extends Cw1CosmWasmClient {
+  private async setAdmins(admins: readonly string[], memo = ""): Promise<ExecuteResult> {
+    const handleMsg = {
+      update_admins: {
+        admins: admins,
+      },
+    };
+    return this.execute(this.cw1ContractAddress, handleMsg, memo);
+  }
+
+  public async getAdmins(): Promise<readonly string[]> {
+    const { admins } = await this.queryContractSmart(this.cw1ContractAddress, { admin_list: {} });
+    return admins;
+  }
+
+  public async isAdmin(address = this.senderAddress): Promise<boolean> {
+    const admins = await this.getAdmins();
+    return admins.includes(address);
+  }
+
+  public async getAllAllowances(): Promise<readonly AllowanceResult[]> {
+    const { allowances } = await this.queryContractSmart(this.cw1ContractAddress, {
```

Comment:
All allowances supports pagination: https://github.com/CosmWasm/cosmwasm-plus/blob/master/contracts/cw1-subkeys/src/msg.rs#L70-L73

This will only ever get the first page (or default length 10).

Please expose pagination in the API, or internally loop to read all allowances in multiple queries before returning.